operator/pkg/nodepools/pools.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package nodepools exists currently because it is required to be a separate due to import conflicts.
```

Comment:
Please specify the exact import conflict / cycle so we can easily verify when this can be removed.

Reply:
added comment